sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2855,6 +2863,50 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
         "enumString" -> enumString))
   }
 
+  def unreleasedThreadError(loggingId: String, newAcquiredThreadInfo: String,
+                            AcquiredThreadInfo: String, timeWaitedMs: Long,
+                            stackTraceOutput: String): Throwable = {
+    new SparkException (
+      errorClass = "CANNOT_LOAD_STATE_STORE.UNRELEASED_THREAD_ERROR",
+      messageParameters = Map(
+        "loggingId" -> loggingId,
+        "newAcquiredThreadInfo" -> newAcquiredThreadInfo,
+        "acquiredThreadInfo" -> AcquiredThreadInfo,
+        "timeWaitedMs" -> timeWaitedMs.toString,
+        "stackTraceOutput" -> stackTraceOutput
```

Comment:
Just wonder is it an user-facing error? If so, are you going to show the stack trace to SQL users?

Reply:
I believe cannotLoadStore is a user-facing error, but I'm not sure if unreleasedThreadError should be as well.

This is from a previous conversation:
> These are not about inputs and outputs from the point of end-to-end query. You can imagine that there is an operator which have to retain accumulators over the streaming query's lifetime (I'm over-simplifying but I guess this is more SQL friendly), and we are checkpointing accumulators for that microbatch to the durable (mostly remote) file system.
> 
> RocksDB is what users pick up as local storage on retaining accumulators. Picking up in-memory map is also feasible. It's just that when Spark runs a microbatch, Spark needs to load the state store for specific microbatch (to continue accumulating), which would involve downloading the files from the remote file system, and deserializing them, and loading the accumulators into local storage. Various interactions happen in there which we want to capture these cases with different sub-categories.
> 
> So it's not directly related to the data users read from/write to, but the internal data maintained by the streaming query.